mmcv/runner/hooks/checkpoint.py

```diff
@@ -21,7 +21,7 @@ def __init__(self,
         self.args = kwargs
 
     @master_only
-    def after_train_epoch(self, runner):
+    def after_epoch(self, runner):
```

Comment:
This may not be expected since the model weights before and after the val epoch are the same. Saving the checkpoint in `after_epoch()` results in duplicated checkpoints.

Reply:
@hellock yeah, I dug deeper and found that whole hooks system isn't working as expected because there is no global `after_epoch` callback that will run after all other handlers were called. We need to decide what to do with #276 and then I will change this PR.

Thank you for your time!